### PR TITLE
UGENE-6170 Correct names of document formats in two workflow elements

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

**Review by DmitriiSukhomlinov — DISMISSED** [review]

**Comment by yalgaer** [comment]
Looks good to me. Will approve after nightly tests are finished.

**Review by smlnkcv — COMMENTED** [review]
The "Filter BAM/SAM files" element's';
 It wasn't mentioned in the original issue, but change "Output format" attribute visible value in the same way: format names should be used instead if format IDs.
Actually bam/sam  instead BAM/SAM.

**Comment by yalgaer** [comment]
A nightly test failed with this patch: GUITest_regression_scenarios_test_1918

File: src/plugins/workflow_designer/src/library/ConvertFilesFormatWorker.cpp [hunk 1]
```diff
@@ -101,9 +101,9 @@ QVariantMap getFormatsMap(MapType mapType) {
         }
         if (format->checkFlags(DocumentFormatFlag_SupportWriting) || (BOOLEANS == mapType)) {
             if (BOOLEANS == mapType) {
-                result[fid] = false;
+                result[AppContext::getDocumentFormatRegistry()->getFormatById(fid)->getFormatName()] = false;
```
Comment (yalgaer) [inline src/plugins/workflow_designer/src/library/ConvertFilesFormatWorker.cpp]: Dirty code that will be hard to debug. Could you please cache the value into a local var.
Same for all other files in the PR where you use this long expressions.

File: src/plugins/workflow_designer/src/library/ConvertFilesFormatWorker.cpp [hunk 2]
```diff
@@ -187,7 +187,12 @@ void ConvertFilesFormatWorker::init() {
     inputUrlPort = ports.value(INPUT_PORT);
     outputUrlPort = ports.value(OUTPUT_PORT);
     targetFormat = getValue<QString>(BaseAttributes::DOCUMENT_FORMAT_ATTRIBUTE().getId());
-    excludedFormats = getValue<QString>(EXCLUDED_FORMATS_ID).split(",", QString::SkipEmptyParts);
+    QStringList excludedFormatNames = getValue<QString>(EXCLUDED_FORMATS_ID).split(",", QString::SkipEmptyParts);
+    for (const QString &formatId : AppContext::getDocumentFormatRegistry()->getRegisteredFormats()) {
```
Comment (yalgaer) [inline src/plugins/workflow_designer/src/library/ConvertFilesFormatWorker.cpp]: Add qAsConst

File: src/plugins/workflow_designer/src/library/FilterBamWorker.cpp [hunk 3]
```diff
@@ -239,7 +240,14 @@ Task *FilterBamWorker::tick() {
             setting.outName = getTargetName(url, outputDir);
             setting.inputUrl = url;
             setting.inputFormat = detectedFormat;
-            setting.outputFormat = getValue<QString>(OUT_FORMAT_ID);
+            QString outputFormatName = getValue<QString>(OUT_FORMAT_ID);
+            for (const QString &formatId : AppContext::getDocumentFormatRegistry()->getRegisteredFormats()) {
```
Comment (yalgaer) [inline src/plugins/workflow_designer/src/library/FilterBamWorker.cpp]: qAsConst

File: src/plugins/workflow_designer/src/library/FilterBamWorker.h [hunk 4]
```diff
@@ -78,6 +78,7 @@ class FilterBamWorkerFactory : public DomainFactory {
     Worker *createWorker(Actor *a) {
         return new FilterBamWorker(a);
     }
+    QMap<QString, QString> formatNameIdAssociation;
```
Comment (yalgaer) [inline src/plugins/workflow_designer/src/library/FilterBamWorker.h]: Please document the field

**Review by yalgaer — CHANGES_REQUESTED** [review]

**Review by yalgaer — APPROVED** [review]

**Pull request merged**